### PR TITLE
enable auto box zoom by default

### LINKED ISSUE #14030: [FEATURE] Box zoom tool - enable dynamic dimensions zooming
### Problem description

Currently the box zoom tool can be configured to handle 2D and 1D zooming. However this be behavior must be set up front and is not supported at run time.

### Feature description

2D zooming (rectangle) should turn into 1D zooming at runtime if the rectangle reaches a very high aspect ratio (close to a vertical or horizontal line). This is especially useful to switch between zooming in x and y directions.
Note: This behavior is implemented in many other popular plotting framework and thus should be considered to be the default.

### Potential alternatives

-

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by muendlein**
This PR enables auto mode for box zoom by default (previously `'both'`), see also https://github.com/bokeh/bokeh/pull/12277
Additionally I have also enhanced the algorithm for switching between `'both'`, `'width'` and `'height'`:

- Increased pixel tolerance from 5 -> 15 to make it easier when selecting a large range
- Introduced an additional aspect ratio tolerance in order to avoid flickering: previously when drawing a box with close to an aspect ratio of 1 it was possible to temporarily switch from `'both'` to `'width' `or `'height'` and then back, e.g. in case of `dx=tol`, `dy=tol-1`

- [x] issues: fixes #14030

### CODE REVIEW

File: bokehjs/src/lib/models/tools/gestures/box_zoom_tool.ts [hunk 1]
```diff
@@ -253,7 +254,7 @@ export class BoxZoomTool extends GestureTool {
     this.prototype.default_view = BoxZoomToolView
 
     this.define<BoxZoomTool.Props>(({Bool, Ref, Or, Auto}) => ({
-      dimensions:   [ Or(Dimensions, Auto), "both" ],
+      dimensions:   [ Or(Dimensions, Auto), "auto" ],
```
Comment (mattpap) [inline bokehjs/src/lib/models/tools/gestures/box_zoom_tool.ts:257]: This change has to be applied in bokeh (Python) as well and then deafults' baseline file regenerated with `pytest tests/test_defaults.py`. I'm surprised this wasn't caught by bokehjs' defaults tests. I will need to investigate this.

**Comment by mattpap** [comment]
> ✗ Tools
> └── should support BoxZoomTool's setup menu
> images differ by 240px (0.46%)

You will have to update the baseline image for this test (see [here](https://docs.bokeh.org/en/latest/docs/dev_guide/testing.html)).

**Comment by muendlein** [comment]
@mattpap Do you have any idea why the test for the baseline image only failed on ubuntu?

**Comment by mattpap** [comment]
> Do you have any idea why the test for the baseline image only failed on ubuntu?

Because we only do image baseline testing on linux/ubuntu.

**Comment by muendlein** [comment]
> > Do you have any idea why the test for the baseline image only failed on ubuntu?
> 
> Because we only do image baseline testing on linux/ubuntu.

Okay understood, now that also means I will have to use linux in order to generate the updated baseline?

**Comment by mattpap** [comment]
Yes, and a very specific setup that matter. Thus, it will be easier to download test artifact from bokejs workflow. It should be explained in the documentation. If not, I will explain it tomorrow.

**Comment by muendlein** [comment]
> Yes, and a very specific setup that matter. Thus, it will be easier to download test artifact from bokejs workflow. It should be explained in the documentation. If not, I will explain it tomorrow.

Just found it, seems like you refer to this https://docs.bokeh.org/en/latest/docs/dev_guide/writing_tests.html#bokehjs-visual-tests (3. Generate CI baselines and commit test). Will do this tomorrow.
Note: The doc here implies that visual testing is also done for MacOS + Windows.

**Comment by mattpap** [comment]
> Note: The doc here implies that visual testing is also done for MacOS + Windows.

We will need to remove this. To clarify, one can run image/visual tests on these platforms, but one can't compare bokehjs' committed baseline images to the generated ones. One would have to first generate local set of baselines, commit them, and then make any relevant changes and re-run tests. We only visual test on Linux to reduce the number of baselines, thus the size of the repository, and reduce testing time.

File: bokehjs/src/lib/models/tools/gestures/box_zoom_tool.ts (hunk 1 above)
Comment (mattpap) [reply bokehjs/src/lib/models/tools/gestures/box_zoom_tool.ts:257]: This is a bug in the runner of defaults' tests.

Comment (mattpap) [reply bokehjs/src/lib/models/tools/gestures/box_zoom_tool.ts:257]: Fixed by PR #14102.

**Review by mattpap — APPROVED** [review]
I think some additional tweaking may be needed to make this even more robust, but for now it's a good enough feature. Thanks @muendlein.

**Pull request merged**